design-collaboration-guidelines.md

```diff
@@ -0,0 +1,82 @@
+
+![typography](./assets/typography.png)
+
+### Layout - Sizing and Spacing
+Defining base templates that respect ratio and size of most used or more common devices that will be used. And a general rule for padding or margin that you can apply to most of the screen.
+
+![layout ios](./assets/layout-ios.png)
+
+![layout android](./assets/layout-android.png)
+
+## Components - Pattern library - Atoms
+Different design systems will have different naming for this, in Atomic Design you will call this Atoms, in others, it's just called Pattern Library, or Components. All these building blocks should be based style guide already provided and will be used to build more complex elements of the user interface like pages and screens.
+
+Commons components  you can find in most applications are:
+* Labels
+* Inputs
+* Buttons
+
+## Interaction Model - Patterns
+
+### Navigation
+As today most of the mobile Apps, are single window full-screen design. In collaboration with the designer, it's essential to review a prototype of the navigation that can be but not mandatory with the final design using tools like Marvel or Invision. There is mainly two kind of Navigation to navigate from screen to another.
+
+Stack, You will find this concept under Navigation Stack or Back Stack under the respective platform. You can quickly identify these screen through the fact that they have a Back/Up Button on the top left Corner.
+Important: Remind UX and Product Owner that navigation from a notification should keep into account the stack designed to get to the desired screen.
+
+![navigation stack](./assets/navigation-stack.png)
+
+Modal/Dialog, It's a screen that sits on top of the current screen, the user must interact with it before they can return to the previous screen. It should contain text and UI controls focused on a specific task. They inform users about critical information, require users to make decisions, or involve multiple tasks. You could identify them as they have at least a cancel button or close button.
+Important: Remind UX and Product Owner a Modal could not be initial screen as they have a dismiss button.
+
+* [Best Practices for Modals / Overlays / Dialog Windows](https://uxplanet.org/best-practices-for-modals-overlays-dialog-windows-c00c66cddd8c)
+
+![navigation stack](./assets/modal.png)
+
+## Empty States
+An essential step while design and estimating how complicated a screen is to define the empty states.
+
+* [Designing For The Empty States](https://tympanus.net/codrops/2013/01/09/designing-for-the-empty-states/)
+* [Material Guidelines Empty States](https://material.io/guidelines/patterns/empty-states.html)
+
+### First use
+It can be the first time that user uses the app, or it can appear until the user makes the first interaction that generates content.
+
+### User Cleared
+Is when the user can clean the content of a screen.
+
+### Error Cases
+Is when the App is not able to load the content of the page.
+
```

Comment:
Adding a picture with an example for these three last points could be really helpful. I've seen designers using different naming conventions for these scenarios, but if you see the picture close to the explanation it could be helpful.